pyanaconda/modules/storage/partitioning/custom/custom_partitioning.py

```diff
@@ -1096,7 +1102,15 @@ def _execute_logvol_data(self, storage, data, logvol_data):
                 maxsize = None
 
             if logvol_data.cache_size and logvol_data.cache_pvs:
-                pv_devices = [lookup_alias(devicetree, pv) for pv in logvol_data.cache_pvs]
+                pv_devices = []
+                for pvname in logvol_data.cache_pvs:
+                    pv = lookup_alias(devicetree, pvname)
+                    if pv.format.type == "luks":
+                        # for encrypted pvs the names are associated with the underlying partition
+                        # but we need the LUKS device on top of it
+                        pv_devices.append(pv.children[0])
+                    else:
+                        pv_devices.append(pv)
```

Comment:
Couldn't you please abstract this code to a method? If I'm looking correctly the code is the same on both places.